src/erlcloud_s3.erl

```diff
@@ -1616,6 +1712,28 @@ s3_xml_request2(Config, Method, Host, Path, Subresource, Params, POSTData, Heade
             Error
     end.
 
+s3_xml_request3(Config, Method, Host, Path, SubResource, Params, POSTData,
+                Headers) ->
+    case s3_request2(Config, Method, Host, Path, SubResource, Params,
+                     POSTData, Headers) of
+        {ok, {_, <<>>}} ->
+            ok;
+        {ok, {_, Body}} ->
+            {XML, _} = xmerl_scan:string(binary_to_list(Body)),
+            {ok, XML};
+        {error, {http_error, _, _, Reason}} = Error ->
```

Comment:
I may guess you no like exceptions :)
however why would we introduce a third way of rerun fir these APIs?